tests/jballerina-unit-test/src/test/resources/test-src/record/record_assignability_negative.bal

```diff
@@ -41,3 +41,31 @@ function testClosedRecordAssignabilityNegative() {
     boolean|string|int|record {|boolean|int b;|} r8 = r7;
     boolean|string|int|record {|boolean|int b;|} _ = r7;
 }
+
+type R1 record {
+    string y?;
+};
+
+type R2 record {
+    int x?;
+};
+
+type R3 record {|
+    string y?;
+    int...;
+|};
+
+function testOpenRecordToRecordWithIncompatibleOptionalFieldTyping() {
+    R2 r1 = {x: 1, "y": 10};
+    R1 _ = r1;
+
+    record {| int|string...; |} r2 = {};
```

Comment:
```suggestion
    record {|int|string...;|} r2 = {};
```
Here and below

Reply:
Same as https://github.com/ballerina-platform/ballerina-lang/pull/40074/files#r1271727926.

Reply:
Will change for consistency here though.